src/core/load-fragment.c

```diff
@@ -3939,7 +3941,14 @@ int config_parse_bind_paths(
                         return 0;
                 }
 
-                s = source;
+                r = unit_full_printf(u, source, &sresolved);
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r,
+                                   "Failed to resolved specifiers in \"%s\", ignoring: %m", source);
+                        return 0;
```

Comment:
> I'll note that the existing calls to unit_full_printf in this file don't handle -ENOMEM specially

You are right. This is the result of https://github.com/systemd/systemd/commit/19f6d710772305610b928bc2678b9d77fe11e770#diff-cc81d855afbe30015a0e769bbf67567fL102.

I think we can fix it later.